<launch>
    <arg name="drone_name" default="tello" doc="The unique name of the drone"/>
    <arg name="launch_rviz" default="true" doc="Display rviz."/>
    <arg name="use_sim" default="true" doc="Set to true to run in simulation mode"/>
    <arg name="image_save_path" default="$(find tello_llm_ros)/images" doc="Set to true to run in simulation mode"/>
    <arg name="video_save_path" default="$(find tello_llm_ros)/videos" doc="Set to true to run in simulation mode"/>

    <arg name="cmd_vel_timeout" default="0.5" doc="Seconds without cmd_vel before stopping"/>

    <node pkg="rviz" type="rviz" name="map_rviz" args="-d $(find tello_llm_ros)/rviz/odom.rviz" if="$(arg launch_rviz)" />

    <node name="$(arg drone_name)_driver" pkg="tello_llm_ros" type="tello_ros_driver.py" output="screen">
        <param name="drone_name" value="$(arg drone_name)" />
        <param name="use_sim" value="$(arg use_sim)" />
        <param name="cmd_vel_timeout" value="$(arg cmd_vel_timeout)" />
        <param name="image_save_path" value="$(arg image_save_path)" />
        <param name="video_save_path" value="$(arg video_save_path)" />
    </node>

    <!-- sim tf -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="map_to_odom_static" 
        args="0 0 0 0 0 0 map $(arg drone_name)/odom" if="$(arg use_sim)"/>

</launch>